lib/static/new-ui/hooks/useHotkey.ts

```diff
@@ -0,0 +1,170 @@
+        'arrowright': 'ArrowRight'
+    };
+
+    if (codeMap[lowerKey]) {
+        return codeMap[lowerKey];
+    }
+
+    return null;
+}
+
+function parseKey(keyString: string): ParsedKey {
+    const parts = keyString.toLowerCase().split('+');
+    const key = parts[parts.length - 1];
+
+    return {
+        key,
+        code: getKeyCode(key),
+        shift: parts.includes('shift'),
+        meta: parts.includes('meta'),
+        ctrl: parts.includes('ctrl'),
+        alt: parts.includes('alt'),
+        mod: parts.includes('mod')
+    };
+}
+
+function isInputFocused(): boolean {
+    const activeElement = document.activeElement;
+    return activeElement instanceof HTMLInputElement ||
+        activeElement instanceof HTMLTextAreaElement ||
+        activeElement?.getAttribute('contenteditable') === 'true';
+}
+function matchesModifiers(event: KeyboardEvent, parsed: ParsedKey): boolean {
+    const isMac = navigator.platform.toUpperCase().indexOf('MAC') >= 0;
+    const modKey = isMac ? event.metaKey : event.ctrlKey;
+
+    if (parsed.mod) {
+        if (!modKey) {
+            return false;
+        }
+        if (isMac && event.ctrlKey) {
+            return false;
+        }
+        if (!isMac && event.metaKey) {
+            return false;
+        }
+    } else {
+        if (!parsed.meta && event.metaKey) {
+            return false;
+        }
+        if (!parsed.ctrl && event.ctrlKey) {
+            return false;
+        }
+    }
+
+    if (parsed.shift !== event.shiftKey) {
+        return false;
+    }
+    if (parsed.alt !== event.altKey) {
+        return false;
+    }
+    if (parsed.meta && !event.metaKey) {
+        return false;
+    }
+    if (parsed.ctrl && !event.ctrlKey) {
+        return false;
+    }
+
+    return true;
+}
+
+export function useHotkey(
+    keyString: string,
+    callback: () => void,
+    options: UseHotkeyOptions = {}
+): void {
+    const {enabled = true, allowInInput = false} = options;
+
+    const handleKeyDown = useCallback((event: KeyboardEvent): void => {
+        if (!enabled) {
+            return;
+        }
+        if (!allowInInput && isInputFocused()) {
+            return;
+        }
+
+        const parsed = parseKey(keyString);
+
+        if (parsed.code) {
+            if (event.code !== parsed.code) {
+                return;
+            }
+        } else {
+            const eventKey = event.key.toLowerCase();
+            const targetKey = parsed.key.toLowerCase();
+            if (eventKey !== targetKey) {
+                return;
+            }
+        }
+
+        if (!matchesModifiers(event, parsed)) {
+            return;
+        }
+
+        event.preventDefault();
+        callback();
+    }, [keyString, callback, enabled, allowInInput]);
+
+    useEffect(() => {
+        document.addEventListener('keydown', handleKeyDown);
+        return () => {
+            document.removeEventListener('keydown', handleKeyDown);
+        };
+    }, [handleKeyDown]);
+}
```

Comment:
It's cool that you implemented it yourself without using a third-party solution.